### PR TITLE
MAINT: Ensure Pythran input for lombscargle are contiguous

### LINKED ISSUE #15964: BUG: lombscargle fails if argument is a view
### Describe your issue.

Since #13316 was merged (i.e., in the scipy v1.7.x series), signal.lombscargle has failed if a positional argument is a view.

### Reproducing Code Example

```python
from scipy.signal import lombscargle
import numpy as np

x = np.arange(16, dtype=float)
yc = np.random.randn(16) + 1j * np.random.randn(16)
freqs = np.array([1, 2, 3], dtype=float)
crash = lombscargle(x, yc.imag, freqs)
```

### Error message

```shell
In [15]: run lombscargle_pythran_bug
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
~/work/programs/py/scipy_test/lombscargle_pythran_bug.py in <module>
      5 yc = np.random.randn(16) + 1j * np.random.randn(16)
      6 freqs = np.array([1, 2, 3], dtype=float)
----> 7 crash = lombscargle(x, yc.imag, freqs)

~/miniconda3/envs/py37/lib/python3.7/site-packages/scipy/signal/spectral.py in lombscargle(x, y, freqs, precenter, normalize)
    148         pgram = _lombscargle(x, y - y.mean(), freqs)
    149     else:
--> 150         pgram = _lombscargle(x, y, freqs)
    151
    152     if normalize:

TypeError: Invalid call to pythranized function `_lombscargle(float64[:], float64[:] (is a view), float64[:])'
Candidates are:

    - _lombscargle(float64[:], float64[:], float64[:])
```

### SciPy/NumPy/Python version information

1.7.3 1.21.5 sys.version_info(major=3, minor=7, micro=12, releaselevel='final', serial=0)

### PR DESCRIPTION
**Pull request by serge-sans-paille**
Fix #15964

### CODE REVIEW

File: scipy/signal/_spectral_py.py [hunk 1]
```diff
@@ -134,9 +134,9 @@ def lombscargle(x,
     >>> plt.show()
 
     """
-    x = np.asarray(x, dtype=np.float64)
-    y = np.asarray(y, dtype=np.float64)
-    freqs = np.asarray(freqs, dtype=np.float64)
+    x = np.ascontiguousarray(x, dtype=np.float64)
```
Comment (thomasjpfan) [inline scipy/signal/_spectral_py.py:137]: I think this makes a copy for non-contiguous arrays. Is this a memory regression compared to the Cython version?

https://github.com/scipy/scipy/blob/2b526a9d2f0a85c804a2363e06111bd6976795e9/scipy/signal/_spectral.pyx#L21-L23

For the Cython version, there is no requirement for contiguous arrays.

**Comment by serge-sans-paille** [comment]
To avoid the copy, we can also change the pythran signature to

float[] or float[:]

for each parameter. I'd favor that one... any preference?

**Review by tupui — APPROVED** [review]
Thank you @serge-sans-paille. From the discussion this seems good. Failure in CI is not related and observed elsewhere. Something broke in NumPy.

**Comment by tylerjereddy** [comment]
Labeled as bug and cleanly cherry-picks to maintenance branch so tentatively labeled for backport.

**Pull request merged**